File: Math & Game Theory & Simulation/Minimum Number of Operations to Make All Array Elements Equal to 1/solution.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        n = len(nums)
        ones = nums.count(1)
        if ones: return n - ones
        result = math.inf
        for i in range(n):
            gcd = nums[i]
            for j in range(i+1, n):
                gcd = math.gcd(gcd, nums[j])
                if gcd == 1:
                    result = min(result, j - i - 1 + n)
        if result == math.inf: return -1
        return result
            

File: Math & Game Theory & Simulation/Minimum Number of Operations to Make All Array Elements Equal to 1/solution.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        int ones = count(nums.begin(), nums.end(), 1);
        if(ones) return nums.size() - ones;
        int result = INT_MAX;
        for(int i = 0; i < nums.size(); ++i) {
            int g = nums[i];
            for(int j = i+1; j < nums.size(); ++j) {
                g = gcd(g, nums[j]);
                if(g == 1) {
                    result = min(result, j - i - 1 + static_cast<int>(nums.size()));
                }
            }
        }
        return result == INT_MAX ? -1: result;
    }
};
